### PR TITLE
Kev/342_correlation_test

### PR DESCRIPTION
**Pull request by junhaow1**
# Pull Request Details

## What issue does this PR address

- TESTING: Demo/Large Explore Correlation #342

- Link to associated issue: #342 

## Checklist

Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [x] Screenshots included in linked issue
- [x] Changes adhere to the style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint check errors related to your changes (`make prep` or `flutter analyze lib`)
- [ ] Pre-exisiting lint errors noted: [HERE]
- [x] Tested on at least one device
  - [ ] Android Phone
  - [ ] Android Emulator
  - [ ] Chrome on Android
  - [ ] Chrome
  - [ ] iOS
  - [ ] Linux
  - [x] MacOS
  - [ ] Windows
- [ ] Added a reviewer

## Finalising

Once PR discussion is complete and reviewer has approved:

- [ ] Merge dev into the branch
- [ ] Resolve any conflicts
- [ ] Add one line summary into CHANGELOG.md
- [ ] Bump appropriate version number in pubspec.yaml
- [ ] Push to git repository and review
- [ ] Merge PR into dev (gjwgit)

### CODE REVIEW

File: integration_test/explore_correlation_demo_test.dart [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import 'package:flutter/material.dart';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:rattle/features/correlation/panel.dart';
+
+import 'package:rattle/main.dart' as app;
+import 'package:rattle/features/dataset/button.dart';
+import 'package:rattle/features/dataset/popup.dart';
+import 'package:rattle/tabs/explore.dart';
```
Comment (gjwgit) [inline integration_test/explore_correlation_demo_test.dart]: Is this required?

File: integration_test/explore_correlation_demo_test.dart [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <https://www.gnu.org/licenses/>.
+///
+/// Authors:  Kevin Wang
+
+library;
+
+// Group imports by dart, flutter, packages, local. Then alphabetically.
+
+import 'package:flutter/material.dart';
```
Comment (gjwgit) [inline integration_test/explore_correlation_demo_test.dart]: Is this used?

**Review by gjwgit — CHANGES_REQUESTED** [review]
The support functions are great but could they each go into a separate file of the name of the function and then import where used?  Perhaps they go under `integration_test/utils` Thanks.

Perhaps navigateToExploreTab() could be generalized to navigateToTab('Explore")?

And naviagateToTab() could be navigateToFeature() for more consistent nomenclature.

THen add a new issue to fix up ALL tests to uses the support functions like loading the DEMO and loading the large.

Thanks.

File: integration_test/explore_correlation_demo_test.dart [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      await tester.pump(pause);
+
+      await _openDemoDataset(tester);
+      await navigateToExploreTab(tester);
+
+      // Use the helper functions from helper.dart
+      await navigateToTab(tester, 'Correlation', CorrelationPanel);
+
+      await performCorrelationAnalysis(tester);
+
```
Comment (gjwgit) [inline integration_test/explore_correlation_demo_test.dart]: Could you add some pauses here for the itest so the tester can see what is going on. THanks.

Comment (junhaow1) [reply integration_test/explore_correlation_demo_test.dart]: Hi @gjwgit  I've made some changes , please have a look at the latest commit. Thanks :)

File: integration_test/explore_correlation_demo_test.dart (hunk 2 above)
Comment (junhaow1) [reply integration_test/explore_correlation_demo_test.dart]: deleted

File: integration_test/explore_correlation_demo_test.dart (hunk 1 above)
Comment (junhaow1) [reply integration_test/explore_correlation_demo_test.dart]: deleted

File: integration_test/utils/oepn_demo.dart
Comment (gjwgit) [inline integration_test/utils/oepn_demo.dart:1]: Misspelled file name.

File: integration_test/explore_correlation_demo_test.dart [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+/// Authors:  Kevin Wang
+
+library;
+
+// Group imports by dart, flutter, packages, local. Then alphabetically.
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:rattle/features/correlation/panel.dart';
+
```
Comment (gjwgit) [inline integration_test/explore_correlation_demo_test.dart:30]: The ordering and gap here is not in compliance.

File: integration_test/utils/verify_page_content.dart [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+// details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program.  If not, see <https://www.gnu.org/licenses/>.
+///
+/// Authors: Kevin Wang
+library;
```
Comment (gjwgit) [inline integration_test/utils/verify_page_content.dart:26]: I think you linter might be deleting a blank line before the library - I am regularly putting it back in there. Could you check this and if so see how to disable that behaviour. Thanks.

**Review by gjwgit — COMMENTED** [review]
Reviewing code. Will push some updates.

**Review by gjwgit — APPROVED** [review]

**Pull request merged**